addons/mail/sfu_server/src/utils/utils.js

```diff
@@ -0,0 +1,167 @@
+        }
+        switch (logLevel ?? config.LOG_LEVEL) {
+            case "none":
+                this.error = () => {};
+            // eslint-disable-next-line no-fallthrough
+            case "error":
+                this.warn = () => {};
+            // eslint-disable-next-line no-fallthrough
+            case "warn":
+                this.info = () => {};
+            // eslint-disable-next-line no-fallthrough
+            case "info":
+                this.debug = () => {};
+            // eslint-disable-next-line no-fallthrough
+            case "debug":
+                this.verbose = () => {};
+        }
+    }
+    /**
+     * @returns {string} a date of this format: `yyyy-mm-ddT hh:mm:ss,mmmZ`
+     */
+    _getTimeStamp() {
+        const now = new Date();
+        return now.toISOString() + " ";
+    }
+    /**
+     * @param {Function} logFn The function used to log the message, e.g. `console.error`
+     * @param {string} level formatted level, e.g. `:ERROR:`
+     * @param {string} text
+     * @param {string} [color]
+     */
+    _log(logFn, level, text, color) {
+        logFn(
+            `${this._getTimeStamp()}discuss-sfu ${this._colorize(
+                `${level} [${this._name}] - ${text}`,
+                color
+            )}`
+        );
+    }
+    /**
+     * @param {string} text
+     * @param {string} [color]
+     * @returns {string}
+     */
+    _colorize(text, color = "") {
+        return `${color}${text}${ASCII.color.default}`;
+    }
+    error(text) {
+        this._log(console.error, `:ERROR:`, text, ASCII.color.red);
+    }
+    warn(text) {
+        this._log(console.error, `:WARN:`, text, ASCII.color.yellow);
+    }
+    info(text) {
+        this._log(console.log, `:INFO:`, text, ASCII.color.green);
+    }
+    debug(text) {
+        this._log(console.log, `:DEBUG:`, text);
+    }
+    verbose(text) {
+        this._log(console.log, `:VERBOSE:`, text, ASCII.color.white);
+    }
+}
+
+/**
+ * @param {import("node:http").IncomingMessage} req
+ * @param {Object} [options]
+ * @param {boolean} [options.json] if the response has to be JSON parsed
+ * @returns {Promise<any>} a string, or any json serializable if option.json = true
+ */
+export function parseBody(req, { json } = {}) {
+    return new Promise((resolve) => {
+        const rawBody = [];
+        req.on("data", (chunk) => {
+            rawBody.push(chunk);
+        });
+        req.on("end", () => {
+            const stringBody = Buffer.concat(rawBody).toString();
+            resolve(json ? JSON.parse(stringBody) : stringBody);
+        });
+    });
+}
+
+/**
+ * @param {import("node:http").IncomingMessage} req
+ * @returns {{ host: string, pathname: string, protocol: string, remoteAddress: string, searchParams: URLSearchParams }}
+ */
+export function extractRequestInfo(req) {
+    const url = new URL(req.url, "http://localhost/"); // the second argument is required to parse the url correctly, but not relevant.
```

Comment:
Is it ok to hard code localhost? Should'nt it be odoo.com on odoo.com?

Reply:
yes it is just required by the URL constructor to create a valid URL object, but I do not rely on that part, it is only used to extract the pathname and the search params